### PR TITLE
Updated foundations for the UI refresh

### PR DESCRIPTION
**Pull request by lawik**

### CODE REVIEW

**Review by nshoes — COMMENTED** [review]
This generally LGTM! Just a few questions.

File: assets/vendor/topbar.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
+/**
```
[the remark this answers is not in the document]
Comment (lawik) [reply assets/vendor/topbar.js:1]: I believe @joshk picke this up from what was already there or how Phoenix does it but not sure. I grabbed it from his PR.

File: lib/nerves_hub_web/components/core_components.ex [hunk 2]
```diff
@@ -0,0 +1,618 @@
+defmodule NervesHubWeb.CoreComponents do
```
[the remark this answers is not in the document]
Comment (lawik) [reply lib/nerves_hub_web/components/core_components.ex:1]: We get no credit. This is what Phoenix generates in modern projects :)

File: lib/nerves_hub_web/components/navigation.ex [hunk 3]
```diff
@@ -7,6 +7,113 @@ defmodule NervesHubWeb.Components.Navigation do
...
+    ~H"""
+    <nav class="flex flex-1 flex-col">
+      <ul role="list" class="flex flex-1 flex-col">
+        <li>
+          <ul role="list">
+            <.nav_link
+              label="Devices"
+              path={~p"/org/#{@org.name}/#{@product.name}/devices"}
+              current={@current}>
+              <path
```
[the remark this answers is not in the document]
Comment (lawik) [reply lib/nerves_hub_web/components/navigation.ex:25]: We are using the icons from the agency's design, not sure if they are from some specific pack.

**Comment by lawik** [comment]
Need to check in on CI but otherwise will merge this shortly and we can start divvying up the work and iterating more :)

File: lib/nerves_hub_web/components/navigation.ex (hunk 3 above)
[the remark this answers is not in the document]
Comment (nshoes) [reply lib/nerves_hub_web/components/navigation.ex:25]: Oh apologies, I thought you referenced Heroicons somewhere. All good if they're from design 👍

File: assets/vendor/topbar.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (nshoes) [reply assets/vendor/topbar.js:1]: Cool. It's not a blocker, but I'd rather consume the package if we aren't modifying it.

**Review by nshoes — APPROVED** [review]

**Pull request merged**